doc/user/upgrade_guide.md

```diff
@@ -0,0 +1,69 @@
+# Upgrade Guide
+
+This document shows how to safely upgrade the operator to a desired version while preserving the cluster's state and data whenever possible. It is assumed that the preexisting cluster is configured to create and store backups to persistent storage. See the [backup config guide](/doc/user/backup_config.md) for details.
+
+### Backup Safety Precaution:
+Before attempting to upgrade the operator it is strongly recommended that a backup is created for the current cluster. This will provide a way to recover the cluster back to its previous state in the case of failures in the upgrade process.
+
+For operator versions that are not backwards compatible with clusters created by previous versions, creating a backup will be necessary in order to recreate the cluster for the upgraded operator. 
+
+To see how to create a backup for an existing cluster see the [backup service guide](/doc/user/backup_service.md), and to restore from a previous backup see the [spec examples](https://github.com/coreos/etcd-operator/blob/master/doc/user/spec_examples.md#three-members-cluster-that-restores-from-previous-pv-backup).
+
+### In-place update
+In order to make [in-place updates](https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/#in-place-updates-of-resources) to the operator deployment image via `kubectl apply` or `kubectl edit` your kubernetes cluster and kubectl versions must be 1.6 or greater.
+
+For kubernetes versions less than 1.6 you will have to get the deployment, edit the image field and use `kubectl replace` to update the operator.
+
+## v0.2.7
```

Comment:
See https://github.com/coreos/etcd-operator/issues/1046 . Our next release will be 0.3.0
For 0.3.x , it's the same.
From 0.4.x on,  we will probably move "upgrade notes" in CHANGELOG to here to keep track of upgrade.